core/store/src/flat_state.rs

```diff
@@ -439,8 +444,9 @@ impl FlatStorageState {
 
         if found_final_block {
             let mut store_update = StoreUpdate::new(storage);
-            for delta in deltas_to_apply.drain(..).rev() {
-                delta.apply_to_flat_state(&mut store_update);
+            let mut delta = FlatStateDelta::new();
+            for new_delta in deltas_to_apply.drain(..).rev() {
+                delta.merge(new_delta);
```

Comment:
You need to apply the delta as well?

Reply:
Lol, yes. And I was wondering why I see errors like "staking invariants doesn't hold".